financepayment/controllers/front/response.php

```diff
@@ -640,4 +632,146 @@ protected function createSignature($payload, $secret)
 
         return $signature;
     }
+
+    /**
+     * checks the integrity and validity of the webhook payload
+     *
+     * @param string $input the raw string of the body
+     * @return object the json decoded webhook payload object
+     * @throws WebhookException if json could not be decoded
+     * @throws WebhookException if secret set but no HMAC received in the header
+     * @throws WebhookException if divido HMAC does not match HMAC generated with config shared secret
+     * @throws WebhookException if a required element of the payload is missing
+     */
+    protected function validateWebhook($input){
```

Comment:
can we declare the `$input` parameter & return type here?
```suggestion
    protected function validateWebhook(string $input): object {
```

same for other methods in this class

Reply:
😬 We've traditionally omitted this for compatibility. Probably safe to assume everyone will at least be using PHP 7 at this point though!